tritium-metrics/src/main/java/com/palantir/tritium/metrics/TaggedMetricsExecutorService.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.tritium.metrics;
+
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+final class TaggedMetricsExecutorService implements ExecutorService {
```

Comment:
If we had a [WrappingExecutorService](https://github.com/palantir/tracing-java/blob/develop/tracing/src/main/java/com/palantir/tracing/WrappingExecutorService.java) available, we could simplify these to call `submitted.mark()` in the callable ctor, and avoid some complexity in each method.
This would result in n invocations of mark(1) rather than mark(n), but I can't imagine that makes a great deal of difference.

Unfortunately we don't have access to that sort of executor, and I'm not sure it's worth defining another two copies in tritium just for these executor implementations.

blab: Ignore this comment, recording for posterity